### PR TITLE
FIX: Exception Handling in AWS Glue renameTable Method

### PR DESCRIPTION
**Pull request by jishangarg**
Related to issue: #11155

This PR addresses an issue in the AWS Glue renameTable method where an incorrect exception is thrown when a table with the new name already exists. According to the Iceberg Catalog interface, the method should throw org.apache.iceberg.exceptions.AlreadyExistsException. However, it currently throws software.amazon.awssdk.services.glue.model.AlreadyExistsException, leading to inconsistent behavior.

Problem:
The renameTable method throws an AWS Glue-specific AlreadyExistsException instead of the expected Iceberg-specific AlreadyExistsException. This violates the contract defined by the Iceberg Catalog interface and can cause confusion for applications relying on the Iceberg exception handling.

Solution:
Updated the glue.createTable() call: The AlreadyExistsException from AWS Glue is now caught and rethrown as Iceberg’s org.apache.iceberg.exceptions.AlreadyExistsException. Ensures that the Iceberg exception is consistently used in line with the catalog interface specification.

Changes:
Wrapped the glue.createTable() call in a try-catch block. Caught software.amazon.awssdk.services.glue.model.AlreadyExistsException and rethrew it as org.apache.iceberg.exceptions.AlreadyExistsException with an appropriate error message. Retained the existing rollback logic for handling failures when dropping the old table.

Impact:
Ensures consistent exception handling in the renameTable method. Aligns the behavior with the Iceberg API contract. No impact on backward compatibility, as this only corrects exception handling. This PR ensures that the renameTable operation behaves as expected according to the Iceberg documentation and resolves the mismatch in exception handling.

### CODE REVIEW

**Comment by jishangarg** [comment]
Can somebody please review my PR ?

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java [hunk 1]
```diff
@@ -415,15 +415,19 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
...
+        glue.createTable(
+            CreateTableRequest.builder()
+                .catalogId(awsProperties.glueCatalogId())
+                .databaseName(toTableDbName)
+                .tableInput(tableInputBuilder.name(toTableName).build())
+                .build());
+        LOG.info("created rename destination table {}", to);
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+    // Catch AWS Glue exception and rethrow it as Iceberg's AlreadyExistsException
+    throw new org.apache.iceberg.exceptions.AlreadyExistsException("Table already exists: " + toTableName, e);
```
Comment (amogh-jahagirdar) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java:428]: Can we use the fully qualified `AlreadyExistsException` for glue and then just use the normal class name the Iceberg `AlreadyExistsException`. That aligns with what's done in the other parts of the code and is a bit more readable imo.

**Review by amogh-jahagirdar — CHANGES_REQUESTED** [review]
Thanks for the fix @jishangarg, overall the fix looks good to me but I think we need a test for this. Could we add one to `TestGlueCatalog` please?

**Comment by amogh-jahagirdar** [comment]
Also double check spotless, seems like those checks are failing. You can run `./gradlew :iceberg-aws:spotlessApply`.
Also cc @geruh @singhpk234 @rahil-c @jackye1995  in case they wanted to take a look

**Comment by singhpk234** [comment]
The particular case referred here makes sense but was thinking if this is something we need to handle broadly ?
IMHO we are lacking the exception handling entirely here for ex: 
1/ What if it's 403 forbidden ? let;s say we have read only permission
2/ What if it;s service unavailable or rate limit exceeded etc 

@amogh-jahagirdar was thinking if we can re-use / generalize this code or classification [1] and wire it to all all glue-clients request / response interaction, your thoughts ?

https://github.com/apache/iceberg/blob/main/aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java#L353

**Comment by amogh-jahagirdar** [comment]
Thanks @singhpk234!

At the moment the [catalog API for renameTable](https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L315) does not specify what the expected exceptions are for the 403 or throttling but I agree that we may as well make those Iceberg exceptions in the catalog implementation. 

I wouldn't be opposed to upleveling the exception handling logic in `GlueTableOperations`  to `GlueToToIcebergConverter` and then using that logic where relevant.

**Pull request closed without merging**